di/src/main/java/com/globant/KoinModules.kt

```diff
@@ -4,18 +4,21 @@ import com.globant.data.database.CharacterDatabase
 import com.globant.data.repositories.MarvelCharacterRepositoryImpl
 import com.globant.data.service.MarvelCharactersServiceImpl
 import com.globant.domain.repositories.MarvelCharacterRepository
+import com.globant.domain.services.MarvelCharactersService
 import com.globant.domain.usecases.GetCharacterByIdUseCase
-import com.globant.domain.usecases.GetCharactersUseCase
+import com.globant.domain.usecases.GetCharacterUseCase
+import com.globant.domain.usecases.implementations.GetCharacterByIdUseCaseImpl
+import com.globant.domain.usecases.implementations.GetCharactersUseCaseImpl
 import org.koin.dsl.module
 
 val repositoriesModule = module {
     single { MarvelCharactersServiceImpl() }
     single { CharacterDatabase() }
     single<MarvelCharacterRepository> { MarvelCharacterRepositoryImpl(get(), get()) }
+    single<MarvelCharactersService> { MarvelCharactersServiceImpl() }
 }
 
-
 val useCasesModule = module {
-    single { GetCharacterByIdUseCase() }
-    single { GetCharactersUseCase() }
+    single<GetCharacterByIdUseCase> { GetCharacterByIdUseCaseImpl() }
+    single<GetCharacterUseCase> { GetCharactersUseCaseImpl() }
 }
```

Comment:
it's necessary for all use cases to be single?

Reply:
No, in next commit i change this, thanks!